File: Backend/SoftAseg/SoftAsegBusiness/src/main/java/pe/edu/pucp/softaseg/bo/PolizaSOATBO.java

package pe.edu.pucp.softaseg.bo;

import java.util.ArrayList;
import pe.edu.pucp.softaseg.dao.PolizaSOATDAO;
import pe.edu.pucp.softaseg.daoImp.PolizaSOATDAOImpl;
import pe.edu.pucp.softaseg.model.gestionSiniestros.PolizasSOATDTO;

public class PolizaSOATBO {
    private PolizaSOATDAO polizaSOATDAO;
    
    public PolizaSOATBO(){
        this.polizaSOATDAO = new PolizaSOATDAOImpl();
    }
    
    public Integer insertar(PolizasSOATDTO polizasSOATDTO){
        Integer polizaId = this.polizaSOATDAO.insertar(polizasSOATDTO);
        polizasSOATDTO.setPolizaId(polizaId);
        return polizaId;
    }
    
    public PolizasSOATDTO obtenerPorId(Integer polizaId){
        PolizasSOATDTO polizasSOATDTO = new PolizasSOATDTO();
        polizasSOATDTO.setPolizaId(polizaId);
        return this.polizaSOATDAO.obtenerPorId(polizaId);
    }
    
    public ArrayList<PolizasSOATDTO> listarTodos(){
        return this.polizaSOATDAO.listarTodos();
    }
    
    public ArrayList<PolizasSOATDTO> listarPorNombreApellidoAsegurado(String filtro){
        return this.polizaSOATDAO.listarPorNombreApellidoAsegurado(filtro);
    }
    
    public Integer modificar(PolizasSOATDTO polizasSOATDTO){
        
        return this.polizaSOATDAO.modificar(polizasSOATDTO);
    }
    
    public Integer eliminar(Integer polizaId){
        PolizasSOATDTO polizasSOATDTO = new PolizasSOATDTO();
        polizasSOATDTO.setPolizaId(polizaId);
        return this.polizaSOATDAO.eliminar(polizasSOATDTO);
    }
}

File: Frontend/SoftSiniNet/SiniNetBusiness/PolizaSOATBO.cs
using SiniNetBusiness.SiniNetWSPolizasSOAT;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SiniNetBusiness
{
    public class PolizaSOATBO
    {
        private PolizasSOATClient clienteSOAP;

        public PolizaSOATBO()
        {
            this.clienteSOAP = new PolizasSOATClient();
        }

        public int insertar(
            personasDTO asegurado,
            estadoPoliza estado,
            int montoACobertura)
        {
            return this.clienteSOAP.insertarPolizasSOAT(asegurado, estado, montoACobertura);
        }

        public polizasSOATDTO obtenerPorId(int polizaId)
        {
            return this.clienteSOAP.obtenerPorIdPolizasSOAT(polizaId);
        }

        public IList<polizasSOATDTO> listarTodos()
        {
            return this.clienteSOAP.listarTodosPolizasSOAT();
        }

        public IList<polizasSOATDTO> listarPorNombreApellidoAsegurado(string filtro)
        {
            return this.clienteSOAP.listarPorNombreApellidoAseguradoPolizasSOAT(filtro);
        }

        public int modificar(
            int polizaId,
            personasDTO asegurado,
            estadoPoliza estado,
            int montoACobertura)
        {
            return this.clienteSOAP.modificarPolizasSOAT(polizaId, asegurado, estado, montoACobertura);
        }

        public int eliminar(int polizaId)
        {
            return this.clienteSOAP.eliminarPolizasSOAT(polizaId);
        }
    }

}
